From 7599cae15fe11a5cd07a029657af294c3ef83fff Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Wed, 15 May 2019 20:13:20 +0200
Subject: [PATCH 1/1] efi_selftest: SetVariable() does not yet implement append

U-Boot does not yet implementing EFI_VARIABLE_APPEND_WRITE for
SetVariable(). In case of an error just write a TODO message.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_selftest/efi_selftest_variables.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/lib/efi_selftest/efi_selftest_variables.c b/lib/efi_selftest/efi_selftest_variables.c
index b028c64bbc..e8346d0d4a 100644
--- a/lib/efi_selftest/efi_selftest_variables.c
+++ b/lib/efi_selftest/efi_selftest_variables.c
@@ -115,10 +115,8 @@ static int execute(void)
 				    EFI_VARIABLE_BOOTSERVICE_ACCESS |
 				    EFI_VARIABLE_APPEND_WRITE,
 				    7, v + 8);
-	if (ret != EFI_SUCCESS) {
-		efi_st_error("SetVariable failed\n");
-		return EFI_ST_FAILURE;
-	}
+	if (ret != EFI_SUCCESS)
+		efi_st_todo("SetVariable: append failed\n");
 	len = EFI_ST_MAX_DATA_SIZE;
 	ret = runtime->get_variable(L"efi_st_var1", &guid_vendor1,
 				    &attr, &len, data);
-- 
2.20.1

